digest.go

```diff
@@ -101,26 +101,18 @@ func FromString(s string) Digest {
 // error if not.
 func (d Digest) Validate() error {
 	s := string(d)
-
 	i := strings.Index(s, ":")
-
-	// validate i then run through regexp
-	if i < 0 || i+1 == len(s) || !DigestRegexpAnchored.MatchString(s) {
+	if i < 0 {
+		return ErrDigestInvalidFormat
+	}
+	algorithm, encoded := Algorithm(s[:i]), s[i+1:]
+	if algorithm == "" || encoded == "" {
```

Comment:
The suggestion was instead of doing this here, when the algorithm is not available, do `!DigestRegexpAnchored.MatchString(s)` and return `ErrDigestInvalidFormat` if it does not match instead of `ErrDigestUnsupported`